src/plugins/auto/plugin.cpp

```diff
@@ -595,4 +596,24 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::FilterDevice(const st
     }
     return filterDevice;
 }
+std::vector<DeviceInformation> MultiDeviceInferencePlugin::FilterDeviceByNetwork(const std::vector<DeviceInformation>& metaDevices,
+                                                InferenceEngine::CNNNetwork network) {
+    std::vector<DeviceInformation> filterDevice;
+    IE_SUPPRESS_DEPRECATED_START
+    for (auto&item : network.getInputsInfo()) {
+        if (item.second->getPartialShape().is_dynamic()) {
+            for (auto& iter : metaDevices) {
+                if (iter.deviceName.find("CPU") != std::string::npos) {
+                    filterDevice.push_back(iter);
+                    break;
+                }
+            }
+            if (filterDevice.size() == 0)
+                IE_THROW(NotFound) << "No available device for dynamic shape network !";
+            return filterDevice;
+        }
+    }
+    IE_SUPPRESS_DEPRECATED_END
```

Comment:
suggest to use ov::Model::is_dynamic here since OV 1.0 must not support dynamism.

Reply:
can be done in a separate PR

Reply:
changed to ov::Model